src/lib.rs

```diff
@@ -550,13 +576,15 @@ impl<T: Counter> Histogram<T> {
             // Do max value first, to avoid max value updates on each iteration:
             let other_max_index = source.index_for(source.max())
                 .expect("Index for max value must exist");
-            let other_count = source[other_max_index];
+            let other_count = source.count_at_index(other_max_index)
+                .expect("max's index must exist");
```

Comment:
It's somewhat arbitrary whether to use `count_at_index(foo).expect(bar)` or `counts[i]` -- I just did whatever felt reasonable, which seems to have meant "the former, unless it was past my syntactic pain threshold". I'm open to a better heuristic, but it sits well with me to have a compiler-enforced place to say "here's why this access must be safe".

Reply:
Yeah, I lean towards `count_at_index` + `expect` too. It makes it clearer where a panic might happen, and also gives us a somewhat understandable human-readable reason for why we believe it isn't possible.